src/horizon_client.rs

```diff
@@ -1948,4 +2051,39 @@ mod tests {
             "account_debited"
         );
     }
+
+    #[tokio::test]
+    async fn test_get_all_liquidity_pools() {
+        let horizon_client =
+            HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
+
+        let all_liquidity_pools_request = AllLiquidityPoolsRequest::new()
+            .add_native_reserve()
+            .add_alphanumeric4_reserve(
+                "USDC".to_string(),
+                "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5".to_string(),
+            )
+            .set_limit(2);
+
+        let all_liquidity_pools_response = horizon_client
+            .get_all_liquidity_pools(&all_liquidity_pools_request)
+            .await;
+
+        assert!(all_liquidity_pools_response.clone().is_ok());
+    }
```

Comment:
We hebben 4 optionele parameters. Ik had in ieder geval 1 per parameter verwacht in dit geval, en wellicht ook 1 zonder parameter.

Tevens zou ik wat checks toevoegen op de inhoude van de response. Zitten de velden erin dit ik verwacht? Deserialized alles goed? Een `is_ok()` assert is wel erg minimaal